From c9aa98485cd058121bbcefc1cbc55071487b22dd Mon Sep 17 00:00:00 2001
From: Alexander Korsunsky <fat-lobyte9@gmail.com>
Date: Fri, 23 Nov 2012 13:35:23 +0100
Subject: [PATCH] Use nullptr instead of nsnull

nsnull has been removed from libxulrunner. Since we compile in C++11 mode
anyway, we can use nullptr instead.
---
 GnomeKeyring.cpp |    6 +++---
 xpcom_abi.cpp    |    4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/GnomeKeyring.cpp b/GnomeKeyring.cpp
index 811669f..ec25513 100644
--- a/GnomeKeyring.cpp
+++ b/GnomeKeyring.cpp
@@ -139,7 +139,7 @@
 template<class T, void F(T *)>
 class AutoPtr {
   public:
-    AutoPtr() : mPtr(nsnull) { }
+    AutoPtr() : mPtr(nullptr) { }
 
     operator T*() {
       return mPtr;
@@ -449,7 +449,7 @@ class AutoPtr {
 {
   nsCOMPtr<nsILoginInfo> loginInfo = do_CreateInstance(NS_LOGININFO_CONTRACTID);
   if (!loginInfo)
-    return nsnull;
+    return nullptr;
 
   loginInfo->SetPassword(NS_ConvertUTF8toUTF16(found->secret));
 
@@ -587,7 +587,7 @@ NS_IMETHODIMP GnomeKeyring::Init()
                                         getter_AddRefs(prefService));
   if (ret != NS_OK) { return ret; }
 
-  ret = prefService->ReadUserPrefs(nsnull);
+  ret = prefService->ReadUserPrefs(nullptr);
   if (ret != NS_OK) { return ret; }
 
   ret = prefService->GetBranch(kPrefsBranch, getter_AddRefs(pref));
diff --git a/xpcom_abi.cpp b/xpcom_abi.cpp
index 055b0ba..19ebb32 100644
--- a/xpcom_abi.cpp
+++ b/xpcom_abi.cpp
@@ -21,7 +21,7 @@ int main(int argc, char **argv) {
 	nsresult rv;
 
 	nsCOMPtr<nsIServiceManager> servMan;
-	rv = NS_InitXPCOM2(getter_AddRefs(servMan), nsnull, nsnull);
+	rv = NS_InitXPCOM2(getter_AddRefs(servMan), nullptr, nullptr);
 	if (!NS_SUCCEEDED(rv)) return rv;
 
 	nsCOMPtr<nsIXULRuntime> xulrun = do_GetService(XULAPPINFO_SERVICE_CONTRACTID, &rv);
@@ -35,6 +35,6 @@ int main(int argc, char **argv) {
 	if (!NS_SUCCEEDED(rv)) return rv;
 	printf("%s_%s\n", xpcomOs.get(), xpcomAbi.get());
 
-	rv = NS_ShutdownXPCOM(nsnull);
+	rv = NS_ShutdownXPCOM(nullptr);
 	return 0;
 }
-- 
1.7.10

